packages/app/src/utils/oneHotEncoding.ts

```diff
@@ -0,0 +1,35 @@
+import { CoreUpsetData, process } from "@visdesignlab/upset2-core";
+
+export const oneHotEncode = (encodeList: string[], data: CoreUpsetData, empty?: boolean) => {
+    const newColNames: string[] = []
+    const encodedData: CoreUpsetData = structuredClone(data);
+
+    // close the error window
+    if (empty) {
+        return encodedData;
+    }
+
+    // get the unique names of every new column to be added. 
+    //    ex: group1_a, group1_b, group2_a, group2_b, group3_c, ....
+    encodeList.forEach((s) => Object.entries(data.items).forEach(([id, row]) => {
+        let names = `${row[s]}`.split(',');
+        names = names.filter((n) => n !== "")
+        newColNames.push(...names.map((val) => `${s}_${val}`))
```

Comment:
A little clean up to make this easier to follow
```suggestion
        const names = `${row[s]}`
          .split(',');
          .filter((n) => n !== "")
          .map((val) => `${s}_${val}`);
        newColNames.push(...names)
```